apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PatientResourceProvider.java

```diff
@@ -420,6 +436,48 @@ private String partDFieldByMonth(CcwCodebookVariable month) {
     return queryBeneficiariesByIdsWithBeneficiaryMonthlys(ids);
   }
 
+  /**
+   * @param yearMonth the {@link BeneficiaryMonthly#getYearMonth()} value to match against
+   * @param contractId the {@link BeneficiaryMonthly#getPartDContractNumberId()} value to match
+   *     against
+   * @return the count of matching {@link Beneficiary#getBeneficiaryId()} values
+   */
+  @Trace
+  private long queryBeneCountByPartDContractCodeAndYearMonth(
+      LocalDate yearMonth, String contractId) {
+    // Create the query to run.
+    CriteriaBuilder builder = entityManager.getCriteriaBuilder();
+    CriteriaQuery<Long> beneCountCriteria = builder.createQuery(Long.class);
+    Root<BeneficiaryMonthly> beneMonthlyRoot = beneCountCriteria.from(BeneficiaryMonthly.class);
+    beneCountCriteria.select(builder.countDistinct(beneMonthlyRoot));
+    beneCountCriteria.where(
+        builder.equal(beneMonthlyRoot.get(BeneficiaryMonthly_.yearMonth), yearMonth),
+        builder.equal(beneMonthlyRoot.get(BeneficiaryMonthly_.partDContractNumberId), contractId));
+
+    // Run the query and return the results.
+    Optional<Long> matchingBeneCount = Optional.empty();
+    Long beneHistoryMatchesTimerQueryNanoSeconds = null;
+    Timer.Context matchingBeneCountTimer =
+        metricRegistry
+            .timer(
+                MetricRegistry.name(
+                    getClass().getSimpleName(),
+                    "query",
+                    "bene_count_by_year_month_part_d_contract_id"))
+            .time();
+    try {
+      matchingBeneCount =
+          Optional.of(entityManager.createQuery(beneCountCriteria).getSingleResult());
+      return matchingBeneCount.get();
+    } finally {
+      beneHistoryMatchesTimerQueryNanoSeconds = matchingBeneCountTimer.stop();
+      TransformerUtils.recordQueryInMdc(
+          "bene_count_by_year_month_part_d_contract_id",
+          beneHistoryMatchesTimerQueryNanoSeconds,
+          matchingBeneCount.isPresent() ? 1 : 0);
```

Comment:
is this what you want (0 or 1) or is is supposed to be 0 or number of records returned, i.e. the underlying Long value?

Reply:
The number of records returned by the query, which in this case is a `select count(distinct ...) ...`.